### PR TITLE
Change from after_create to after_create_commit

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Running this code on `after_create` creates a situation where the `where` clause in the sidekiq worker cannot yet find the data because the record can’t be seen by other database connections, like the one Sidekiq is using, until the database transaction is committed, which happens a little later.

Basically `after_create` runs before the transaction is actually final and this creates a little race condition.

More info here: https://www.justinweiss.com/articles/a-couple-callback-gotchas-and-a-rails-5-fix/

I think this wasn't a problem for Delayed Job because it always happened after at least one other DB transaction to create the job in the first place.

### CODE REVIEW

**Comment by rhymes** [comment]
It's funny that we didn't incur on this before just because DelayedJob is too slow
